model/src/main/java/io/spine/examples/shareaware/wallet/MoneyReservationSignal.java

```diff
@@ -24,23 +24,45 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-package io.spine.examples.shareaware.wallet.event;
+package io.spine.examples.shareaware.wallet;
 
 import com.google.errorprone.annotations.Immutable;
 import io.spine.annotation.GeneratedMixin;
 import io.spine.base.EventMessage;
+import io.spine.examples.shareaware.OperationId;
 import io.spine.examples.shareaware.WithdrawalId;
 
 /**
- * Common interface for events participating in wallet withdrawal operation.
+ * Common interface for signals participating in the money reservation operation.
  */
 @Immutable
 @GeneratedMixin
-public interface WithdrawalEvent extends EventMessage {
+public interface MoneyReservationSignal extends EventMessage {
 
     /**
-     * Every event that participates in the withdrawal operation
-     * should have {@code WithdrawalId}.
+     * Returns the ID of operation, in the scope of which this signal is emitted.
      */
-    WithdrawalId getWithdrawalProcess();
+    OperationId getOperation();
+
+    /**
+     * Retrieves {@code WithdrawalId} from {@code OperationId} without checking for its existence.
+     *
+     * <p>In case when {@code WithdrawalId} is not existed in {@code OperationId}
+     * this method will return the
+     * <a href="https://protobuf.dev/reference/java/java-generated/#:~:text=static%20Foo%20getDefaultInstance,its%20newBuilderForType()%20method.">
+     * default instance</a> of {@code WithdrawalId}.
+     */
+    default WithdrawalId getWithdrawalProcess() {
+        return getOperation().getWithdrawal();
+    }
+
+    /**
+     * Retrieves the value of {@code WithdrawalId}
+     * without checking for {@code WithdrawalId} existence in {@code OperationId}.
+     *
+     * <p>In case when {@code WithdrawalId} is not existed in {@code OperationId} it will return an empty string.
```

Comment:
"...is not set in the corresponding {@code OperationId}, returns an empty string."